app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,59 @@
+class BlogsController < ApplicationController
+
+  before_action :logged_in_user, only: [:new, :edit, :index, :show, :create,]
```

Comment:
Use devise before_filter authentication on checking login user. Check again the documentation, no need to create your own before_action since it is already supplied by devise.